### PR TITLE
Fix tests on macOS 13

### PR DESCRIPTION
**Pull request by neonichu**
Python is no longer installed in `/usr/bin`, so support `/usr/local` as well. In the medium term, we should not depend on Python for these tests.

### CODE REVIEW

**Comment by neonichu** [comment]
@swift-ci please test

**Comment by neonichu** [comment]
```
[295/295] Testing TSCUtilityTests.ProgressAnimationTests/testNinjaProgressAnimationDumbTerminal
Test Suite 'Selected tests' started at 2022-10-14 18:51:14.028Test Suite 'FileSystemTests' started at 2022-10-14 18:51:14.029Test Case 'FileSystemTests.testLocalReadWriteFile' started at 2022-10-14 18:51:14.029/home/build-user/swift-tools-support-core/Tests/TSCBasicTests/FileSystemTests.swift:340: error: FileSystemTests.testLocalReadWriteFile : XCTAssertEqual failed: ("FileSystemError(kind: TSCBasic.FileSystemError.Kind.isDirectory, path: Optional(<AbsolutePath:"/">))") is not equal to ("FileSystemError(kind: TSCBasic.FileSystemError.Kind.ioError(code: 17), path: Optional(<AbsolutePath:"/">))") -Test Case 'FileSystemTests.testLocalReadWriteFile' failed (0.114 seconds)Test Suite 'FileSystemTests' failed at 2022-10-14 18:51:14.143
```

Interesting, looks like this failure actually isn't macOS 13 specific but also happening on the Linux CI?

**Comment by neonichu** [comment]
@swift-ci please test

**Comment by neonichu** [comment]
I think I am going to undo the `FileSystemError` part of this, not sure why it fails sometimes, but it doesn't necessarily seem to be related to the host OS.

**Comment by neonichu** [comment]
@swift-ci please test

File: Tests/TSCBasicTests/ProcessTests.swift [hunk 1]
```diff
@@ -255,7 +264,7 @@ class ProcessTests: XCTestCase {
 
     func testStdin() throws {
         var stdout = [UInt8]()
-        let process = Process(args: script("in-to-out"), outputRedirection: .stream(stdout: { stdoutBytes in
+        let process = Process(args: script("in-to-out"), environment: env(), outputRedirection: .stream(stdout: { stdoutBytes in
```
Comment (tomerd) [inline Tests/TSCBasicTests/ProcessTests.swift:267]: is there a way to make this the default when not otherwise specified?

**Review by tomerd — APPROVED** [review]
one suggestion

File: Tests/TSCBasicTests/ProcessTests.swift (hunk 1 above)
Comment (neonichu) [reply Tests/TSCBasicTests/ProcessTests.swift:267]: I guess we could create an extension for the purpose of the tests?

**Comment by neonichu** [comment]
@swift-ci please test

**Review by tomerd — APPROVED** [review]

**Pull request merged**